src/test/apiv2.spec.ts

```diff
@@ -71,6 +72,54 @@ describe("apiv2", () => {
       expect(nock.isDone()).to.be.true;
     });
 
+    it("should be able to stream a GET request", async () => {
+      nock("https://example.com")
+        .get("/path/to/foo")
+        .reply(200, "ablobofdata");
+
+      const c = new Client({ urlPrefix: "https://example.com" });
+      const r = await c.request<unknown, NodeJS.ReadableStream>({
+        method: "GET",
+        path: "/path/to/foo",
+        responseType: "stream",
+      });
+      const data = await streamToString(r.body);
+      expect(data).to.deep.equal("ablobofdata");
+      expect(nock.isDone()).to.be.true;
+    });
+
+    it("should throw on a 404 GET request", async () => {
+      nock("https://example.com")
+        .get("/path/to/foo")
+        .reply(404, "not here");
+
+      const c = new Client({ urlPrefix: "https://example.com" });
+      const r = c.request<unknown, NodeJS.ReadableStream>({
+        method: "GET",
+        path: "/path/to/foo",
+        responseType: "stream",
+      });
+      await expect(r).to.eventually.be.rejectedWith(FirebaseError, /Not Found/);
+      expect(nock.isDone()).to.be.true;
+    });
+
+    it("should be able to resolve a stream on a 404 GET request", async () => {
+      nock("https://example.com")
+        .get("/path/to/foo")
+        .reply(404, "does not exist");
+
+      const c = new Client({ urlPrefix: "https://example.com" });
+      const r = await c.request<unknown, NodeJS.ReadableStream>({
+        method: "GET",
+        path: "/path/to/foo",
+        responseType: "stream",
+        resolveOnHTTPError: true,
+      });
+      const data = await streamToString(r.body);
+      expect(data).to.deep.equal("does not exist");
+      expect(nock.isDone()).to.be.true;
+    });
+
```

Comment:
Please add a test for non-404 responses.

Reply:
Line 75 is a streamed GET request with a 200 response.

Reply:
Ah, can we have a 400 error too? I see those errors hit a different branch that deals with response body.